fuzzers/centipede/fuzzer.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration code for centipede fuzzer."""
+
+import subprocess
+import os
+
+from fuzzers import utils
+
+
+def build():
+    """Build benchmark."""
+    # TODO(Dongge): Build targets with sanitizers.
+    cflags = [
+        '-fsanitize-coverage='
+        'trace-pc-guard,'
+        'pc-table,'
+        'trace-cmp,'
+        'trace-loads,'
+        'trace-stores', '-ldl', '-lrt', '-lpthread', '/lib/weak.o'
+    ]
+    utils.append_flags('CFLAGS', cflags)
+    utils.append_flags('CXXFLAGS', cflags)
+
+    os.environ['CC'] = '/bin/clang/bin/clang'
+    os.environ['CXX'] = '/bin/clang/bin/clang++'
+    # TODO(Dongge): Replace the following libs with a single .a file
+    # once Centipede has it
+    os.environ['FUZZER_LIB'] = (
+        '/src/centipede/bazel-bin/libfuzz_target_runner.a '
+        '/src/centipede/bazel-bin/libfuzz_target_runner_no_main.a '
+        '/src/centipede/bazel-bin/libshared_memory_blob_sequence.a '
+        '/src/centipede/bazel-bin/libexecution_request.a '
+        '/src/centipede/bazel-bin/libexecution_result.a '
+        '/src/centipede/bazel-bin/libbyte_array_mutator.a')
+
+    utils.build_benchmark()
+
+
+def fuzz(input_corpus, output_corpus, target_binary):
+    """Run fuzzer. Wrapper that uses the defaults when calling
+    run_fuzzer."""
+    run_fuzzer(input_corpus, output_corpus, target_binary)
+
+
+def run_fuzzer(input_corpus, output_corpus, target_binary, extra_flags=None):
+    """Run fuzzer."""
+    if extra_flags is None:
+        extra_flags = []
+
+    # Seperate out corpus and crash directories as sub-directories of
+    # |output_corpus| to avoid conflicts when corpus directory is reloaded.
+    work_dir = os.path.join(output_corpus, 'WD')
+    work_dir_crash = os.path.join(work_dir, 'crashes')
+    crashes_dir = os.path.join(output_corpus, 'crashes')
+    output_corpus = os.path.join(output_corpus, 'corpus')
+    os.makedirs(work_dir)
+    os.symlink(crashes_dir, work_dir_crash)
+    os.makedirs(crashes_dir)
+    os.makedirs(output_corpus)
+
+    flags = [
+        f'--workdir={work_dir}',
+        f'--corpus_dir={output_corpus},{input_corpus}',
+        f'--binary={target_binary}',
+        '--num_runs=100',
```

Comment:
What does num_runs do?

Reply:
Yeah, it should not be here. Removed : )